src/provider/ContextProvider.js

```diff
@@ -54,11 +55,15 @@ const ContextProvider = ({ children }) => {
       }
 
       const newCheckout = await createNewCheckout()
-      setCheckoutInState(newCheckout)
+      if (!isRemoved) {
+        setCheckoutInState(newCheckout)
+      }
     }
 
     initializeCheckout()
   }, [store.client.checkout])
+  
+  useEffect(() => () => { isRemoved = true; }, [])
```

Comment:
This has some code smell. See here: https://dev.to/juliang/cancelling-a-promise-with-react-useeffect-3062